addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComCurrentMessage.java

```diff
@@ -0,0 +1,213 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rfxcom.internal.messages;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.smarthome.core.library.items.NumberItem;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.rfxcom.RFXComValueSelector;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComException;
+
+/**
+ * RFXCOM data class for current message.
+ *
+ * @author Unknown - Initial contribution
+ * @author Pauli Anttila - for the Similar RFXComEnergyMessage code
+ * @author Jordan Cook - Added support for CURRENT devices, such as OWL CM113
+ */
+public class RFXComCurrentMessage extends RFXComBaseMessage {
+
+    public enum SubType {
+        UNSUPPORTED(0),
+        ELEC1(1),
+        UNKNOWN(255);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        SubType(byte subType) {
+            this.subType = subType;
+        }
+
+        public byte toByte() {
+            return (byte) subType;
+        }
+    }
+
+    private final static List<RFXComValueSelector> supportedInputValueSelectors = Arrays.asList(
+            RFXComValueSelector.SIGNAL_LEVEL, RFXComValueSelector.BATTERY_LEVEL, RFXComValueSelector.CHANNEL_1_AMPS,
+            RFXComValueSelector.CHANNEL_2_AMPS, RFXComValueSelector.CHANNEL_3_AMPS);
+
+    private final static List<RFXComValueSelector> supportedOutputValueSelectors = Arrays.asList();
+
+    public SubType subType = SubType.ELEC1;
+    public int sensorId = 0;
+    public byte count = 0;
+    public double channel1Amps = 0;
+    public double channel2Amps = 0;
+    public double channel3Amps = 0;
+    public byte signalLevel = 0;
+    public byte batteryLevel = 0;
+
+    public RFXComCurrentMessage() {
+        packetType = PacketType.CURRENT;
+    }
+
+    public RFXComCurrentMessage(byte[] data) {
+        encodeMessage(data);
+    }
+
+    @Override
+    public String toString() {
+        String str = "";
+
+        str += super.toString();
+        str += ", Sub type = " + subType;
+        str += ", Device Id = " + getDeviceId();
+        str += ", Count = " + count;
+        str += ", Channel 1 Amps = " + channel1Amps;
+        str += ", Channel 2 Amps = " + channel2Amps;
+        str += ", Channel 3 Amps = " + channel3Amps;
+        str += ", Signal level = " + signalLevel;
+        str += ", Battery level = " + batteryLevel;
+
+        return str;
+    }
+
+    @Override
+    public void encodeMessage(byte[] data) {
+
+        super.encodeMessage(data);
+
+        try {
+            subType = SubType.values()[super.subType];
```

Comment:
Can you bring the retrieving of the enum value in line with how it's resolved since my pull request was merged? So add a static method to the enum.